### PR TITLE
Parse select top in the teradata dialect.

### PR DESCRIPTION
**Pull request by facundominguez**

### CODE REVIEW

File: src/Database/Sql/Type/Query.hs [hunk 1]
```diff
@@ -75,6 +75,12 @@ newtype Distinct = Distinct Bool
 notDistinct :: Distinct
 notDistinct = Distinct False
 
+newtype Top = Top (Maybe Int)
```
Comment (dlthomas-leapyear) [inline src/Database/Sql/Type/Query.hs]: Ah, I'd envisioned reusing the `Limit` constructor.  Lemme chew on this a bit.

Comment (facundominguez) [reply src/Database/Sql/Type/Query.hs]: Ok. BTW, we probably should redefine this as
```
data Top = Top Int | NoTop
```

Comment (facundominguez) [reply src/Database/Sql/Type/Query.hs]: @dlthomas-leapyear I revisited this.

In `queryparser/src/Database/Sql/Type/Query.hs` we have
```haskell
data Query r a
    = QuerySelect a (Select r a)
    | QueryExcept a (ComposedQueryColumns r a) (Query r a) (Query r a)
    | QueryUnion a Distinct (ComposedQueryColumns r a) (Query r a) (Query r a)
    | QueryIntersect a (ComposedQueryColumns r a) (Query r a) (Query r a)
    | QueryWith a [CTE r a] (Query r a)
    | QueryOrder a [Order r a] (Query r a)
    | QueryLimit a (Limit a) (Query r a)
    | QueryOffset a (Offset a) (Query r a)
```

Which assumes that there is a LIMIT query. In Teradata, however, we have tests like:
```haskell
goldenSql "TOP" "SELECT TOP 10 x FROM t"
```
which make `top` a part of the `select` statement, not a query.

The parser of select is
```haskell
selectP :: Parser (Select RawNames Range)
```
Should the type be changed to
```haskell
selectP :: Parser (Query RawNames Range)
```
?

Comment (dlthomas-leapyear) [reply src/Database/Sql/Type/Query.hs]: Ah, yeah, I think you will need to change the type of `selectP` to return a `Query`

**Comment by facundominguez** [comment]
PTAL

**Comment by facundominguez** [comment]
\o/

**Pull request merged**